File: tests/php/testcases/SumTreeTest.php
<?
require_once 'base.php';

class SumTreeTestCase extends TestServiceTestCase {

    public function test_notset() {
        $p = new Tree();
        $r = $this->service->sum_tree($p);
        $this->assertEquals($r, 0);
    }

    public function test_null() {
        $p = new Tree();
        $p->value = null;
        $p->left = null;
        $p->right = null;
        $r = $this->service->sum_tree($p);
        $this->assertEquals($r, 0);
    }

    public function test_value() {
        $p = new Tree();
        $p->value = 10;
        $p->left = null;
        $p->right = null;
        $r = $this->service->sum_tree($p);
        $this->assertEquals($r, 10);
    }

    public function test() {
        $p = new Tree();
        $p->value = 10;
        $p->left = new Tree();
        $p->left->value = 20;
        $p->right = new Tree();
        $p->right->value = 30;
        $r = $this->service->sum_tree($p);
        $this->assertEquals($r, 60);
    }
}

File: tests/mono/testcases/SumTreeTest.cs
using System;
using NUnit.Framework;

namespace TestCases {

    [TestFixture()]
    public class SumTreeTest : TestServiceTest {

        [Test()]
        public void test_null_args() {
            Assert.AreEqual(0, service.sum_tree(null));
        }

        [Test()]
        public void test_notset() {
            Assert.AreEqual(0, service.sum_tree(new Tree()));
        }

        [Test()]
        public void test_null() {
            Tree p = new Tree();
            p.value = null;
            p.left = null;
            p.right = null;
            Assert.AreEqual(0, service.sum_tree(p));
        }

        [Test()]
        public void test_value() {
            Tree p = new Tree();
            p.value = 10;
            p.left = null;
            p.right = null;
            Assert.AreEqual(10, service.sum_tree(p));
        }

        [Test()]
        public void test() {
            Tree l = new Tree();
            l.value = 20;
            Tree r = new Tree();
            r.value = 30;
            Tree p = new Tree();
            p.value = 10;
            p.left = l;
            p.right = r;
            Assert.AreEqual(60, service.sum_tree(p));
        }
    }
}
